analyzer/codechecker_analyzer/buildlog/log_parser.py

```diff
@@ -802,7 +798,27 @@ def parse_options(compilation_db_entry):
     return BuildAction(**details)
 
 
+class CompileCommandEncoder(json.JSONEncoder):
+    """JSON serializer for objects not serializable by default json code"""
+    def default(self, o):
+        if isinstance(o, BuildAction):
+            return o.to_dict()
+        else:
+            # Let the base class default method raise the TypeError
+            return json.JSONEncoder.default(self, o)
+
+
+class CompileActionUniqueingType(object):
+    NONE = 0  # Full Action text
+    SOURCE_ALPHA = 1  # Based on source file, uniqueing by
+    # on alphanumerically first target
+    SOURCE_REGEX = 2  # Based on source file, uniqueing by regex filter
+    STRICT = 3  # Gives error in case of duplicate
+
+
```

Comment:
I think we could rename `parse_log` to reflect that uniqueing is done too.

If multiple compile command json files are given the uniqued version of the last one will overwrite the other uniqued compile command jsons the same name is used as an output: ```uniqued_compilation_db_file = os.path.join(report_dir, "unique_compile_commands.json")```

I would handle the writing out of the unique compile command json file one level up in `analyze.py` where we iterate over multiple compile command json files, this way the `report_dir` is not required as a parameter here.

Please add the new parameter description to the function docstring.